### PR TITLE
1063_derive_param_first_event: implement derive_param_first_event()

### LINKED ISSUE #1063: Feature Request: Implement `derive_param_first_event()`
### Feature Idea

```
derive_param_first_event(
  dataset,
  dataset_adsl,
  dataset_source,
  filter_source,
  date_var,
  subject_keys = vars(STUDYID, USUBJID),
  set_values_to)
```
adds a new parameter to the input dataset. `AVALC` is set to `"Y"` if an observations exists in the input dataset where `PARAMCD == source_param` and `cond == TRUE`. Otherwise `AVALC` is set to `"N"`. `AVAL` is set to `1` or `0`. `ADT` is set to the first date where the condition is fulfilled.

### Relevant Input

What should the input look like? REMINDER: no patient level data or company sensitive information should be shared via this open public issue

### Relevant Output

What should the output look like? REMINDER: no patient level data or company sensitive information should be shared via this open public issue

### Reproducible Example/Pseudo Code

```
derive_param_first_event(
  dataset = adrs,
  dataset_adsl = adsl,
  dataset_source = adrs,
  filter_source = PARAMCD == "OVR" & AVALC == "PD" & ANL01FL=”Y”,
  date_var = ADT,
  subject_keys = vars(STUDYID, USUBJID),
  set_values_to= vars(
    PARAMCD = "PD",
    PARAM = "Disease Progression by Investigator",
    PARCAT1 = "Tumor Response",
    PARCAT2 = "Investigator",
    PARCAT3 = "Recist 1.1",
    ANL01FL = "Y"))

derive_param_first_event(
  dataset = adrs,
  dataset_adsl = adsl,
  dataset_source = adsl,
  filter_source = !is.na(DTHDT),
  date_var = DTHDT,
  set_values_to = vars(
    PARAMCD = "DEATH",
    PARAM = "Death",
    PARCAT1 = "Reference Event",
    ANL01FL = "Y"
  )
)
```

### PR DESCRIPTION
**Pull request by bundfussr**
Closes #1063

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Review by amsmith214 — CHANGES_REQUESTED** [review]
just the one lint warning. everything else looks good!

File: tests/testthat/test-derive_param_first_event.R [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        STUDYID = "XX1234",
+        PARAMCD = "PD",
+        ANL01FL = "Y")
+    )
+
+  expect_dfs_equal(base = expected,
+                   comp = actual,
+                   keys = c("USUBJID", "PARAMCD", "ADT"))
+})
+
```
[the remark this answers is not in the document]
Comment (amsmith214) [reply tests/testthat/test-derive_param_first_event.R:68]: i am not sure why it seems my browser is loading results of the checks slowly or on a lag. apologies. the lint check now passes

**Comment by amsmith214** [comment]
probably just need to merge devel in again, there is a merge conflict in `news.md`

**Comment by bundfussr** [comment]
@amsmith214 , I fixed the merge conflict.

**Review by amsmith214 — APPROVED** [review]

**Pull request merged**